src/render/blog/adapt-service-without-webapi.html.md

```diff
@@ -0,0 +1,505 @@
+
+```json
+{
+  "main": "io.knotx.tutorials.BooksDbAdapter",
+  "options": {
+    "config": {
+      "address": "knotx.adapter.service.custom",
+      "clientOptions": {
+         //we will put database connection options here later
+      }
+    }
+  }
+}
+```
+
+This configuration file is prepared to run the custom _Service Adapter_, starting the
+`io.knotx.tutorials.BooksDbAdapter` _Verticle_ and listening at the address `knotx.adapter.service.custom` on the event bus.
+
+Now we will implement a Java model to read the configuration:
+
+```java
+package io.knotx.tutorials;
+
+import io.vertx.core.json.JsonObject;
+
+public class BooksDbAdapterConfiguration {
+
+  private String address;
+
+  private JsonObject clientOptions;
+
+  public BooksDbAdapterConfiguration(JsonObject config) {
+    address = config.getString("address");
+    clientOptions = config.getJsonObject("clientOptions", new JsonObject());
+  }
+
+  public JsonObject getClientOptions() {
+    return clientOptions;
+  }
+
+  public String getAddress() {
+    return address;
+  }
+}
+```
+
+## Registering a Service Proxy
+
+The next step would be to register an [`AdapterProxy`](https://github.com/Cognifide/knotx/wiki/Adapter#how-to-extend) 
+to handle incoming requests. The simplest way to achieve this is to create a class 
+that extends [`AbstractAdapterProxy`](https://github.com/Cognifide/knotx/blob/master/knotx-core/src/main/java/io/knotx/adapter/AbstractAdapterProxy.java). Let's call it `BooksDbAdapterProxyImpl`:
+
+```java
+package io.knotx.tutorials.impl;
+
+import io.knotx.adapter.AbstractAdapterProxy;
+import io.knotx.dataobjects.AdapterRequest;
+import io.vertx.rxjava.ext.jdbc.JDBCClient;
+import rx.Observable;
+
+public class BooksDbAdapterProxyImpl extends AbstractAdapterProxy {
+
+  //we will need JDBC Client here to perform DB queries
+  private final JDBCClient client;
+
+  public BooksDbAdapterProxyImpl(JDBCClient client) {
+    this.client = client;
+  }
+
+  @Override
+  protected Observable<AdapterResponse> processRequest(AdapterRequest adapterRequest) {
+    // all the Custom Adapter's work will be done here
+  }
+
+}
+```
+
+Now we should register this `AdapterProxy` in the `start()` method of our `BooksDbAdapter` and set it up
+with the following configuration:
+
+```java
+public class BooksDbAdapter extends AbstractVerticle {
+  
+  private MessageConsumer<JsonObject> consumer;
+  private BooksDbAdapterConfiguration configuration;
+
+  @Override
+  public void init(Vertx vertx, Context context) {
+    super.init(vertx, context);
+    // using config() method from AbstractVerticle we simply pass our JSON file configuration to Java model
+    configuration = new BooksDbAdapterConfiguration(config());
+  }
+
+  @Override
+  public void start() throws Exception {
+    //we create RX version of vertx instance to easier handle reactive programming
+    final io.vertx.rxjava.core.Vertx rxVertx = new io.vertx.rxjava.core.Vertx(this.vertx);
```

Comment:
do we need fully qualified package name ? if we used AbstractVerticle class from rxjava package then this.vertx will give use rxified vertx object.